tests/browser/test/delete_blocks_test.js

```diff
@@ -0,0 +1,226 @@
+      // Clear the workspace manually so we can ensure it's clear before moving on to the next test.
+      Blockly.getMainWorkspace().clear();
+    });
+    // Wait for the workspace to be cleared of blocks (no blocks found on main workspace)
+    await browser
+      .$(
+        '.blocklySvg .blocklyWorkspace > .blocklyBlockCanvas > .blocklyDraggable'
+      )
+      .waitForExist({timeout: 2000, reverse: true});
+
+    // Load the start blocks
+    await browser.execute((blocks) => {
+      Blockly.serialization.workspaces.load(blocks, Blockly.getMainWorkspace());
+    }, startBlocks);
+    // Wait for there to be a block on the main workspace before continuing
+    (await getBlockElementById(browser, firstBlockId)).waitForExist({
+      timeout: 2000,
+    });
+  });
+
+  test('Delete block using backspace key', async function () {
+    const before = await getNumberOfBlocks(browser);
+    // Get first print block, click to select it, and delete it using backspace key.
+    const block = (await getBlockElementById(browser, firstBlockId)).$(
+      '.blocklyPath'
+    );
+    await block.click();
+    await browser.keys([Key.Backspace]);
+    const after = await getNumberOfBlocks(browser);
+    chai.assert.equal(
+      before - 2,
+      after,
+      'Expected there to be two fewer blocks after deletion of block and shadow'
+    );
+  });
+
+  test('Delete block using delete key', async function () {
+    const before = await getNumberOfBlocks(browser);
+    // Get first print block, click to select it, and delete it using delete key.
+    const block = (await getBlockElementById(browser, firstBlockId)).$(
+      '.blocklyPath'
+    );
+    await block.click();
+    await browser.keys([Key.Delete]);
+    const after = await getNumberOfBlocks(browser);
+    chai.assert.equal(
+      before - 2,
+      after,
+      'Expected there to be two fewer blocks after deletion of block and shadow'
+    );
+  });
+
+  test('Delete block using context menu', async function () {
+    const before = await getNumberOfBlocks(browser);
+    // Get first print block, click to select it, and delete it using context menu.
+    const block = (await getBlockElementById(browser, firstBlockId)).$(
+      '.blocklyPath'
+    );
+    await contextMenuSelect(browser, block, 'Delete 2 Blocks');
+    const after = await getNumberOfBlocks(browser);
+    chai.assert.equal(
+      before - 2,
+      after,
+      'Expected there to be two fewer blocks after deletion of block and shadow'
+    );
+  });
+
+  test('Undo block deletion', async function () {
+    const before = await getNumberOfBlocks(browser);
+    // Get first print block, click to select it, and delete it using backspace key.
+    const block = (await getBlockElementById(browser, firstBlockId)).$(
+      '.blocklyPath'
+    );
+    await block.click();
+    await browser.keys([Key.Backspace]);
+    await browser.pause(pauseLength);
+    // Undo
+    await browser.keys([Key.Ctrl, 'Z']);
```

Comment:
Question: is ^Z supported on all platforms?  On macOS I'd assume it was ⌘Z.

Reply:
`Key.Ctrl` is platform-specific, so on Mac it will correspond to Command. `Key.Control` is specifically "control"